pytorch_lightning/profiler/pytorch.py

```diff
@@ -277,14 +276,12 @@ def __init__(
         """
         super().__init__(dirpath=dirpath, filename=filename, output_filename=output_filename)
 
-        record_functions = self.__deprecation_check(profiled_functions, record_functions)
-
         self._group_by_input_shapes = group_by_input_shapes and profiler_kwargs.get("record_shapes", False)
         self._emit_nvtx = emit_nvtx
         self._export_to_chrome = export_to_chrome
         self._row_limit = row_limit
         self._sort_by_key = sort_by_key or f"{'cuda' if profiler_kwargs.get('use_cuda', False) else 'cpu'}_time_total"
-        self._user_record_functions = record_functions
+        self._user_record_functions = set() if record_functions is None else record_functions
```

Comment:
```suggestion
        self._user_record_functions = record_functions or set()
```

Reply:
@carmocca ah right, I always forget about that `or` operator. Thanks, I'll include this change in my next contribution 👍🏽